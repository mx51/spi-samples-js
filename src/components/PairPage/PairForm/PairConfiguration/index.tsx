import React from 'react';
import Checkbox from '@material-ui/core/Checkbox';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Grid from '@material-ui/core/Grid';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import Typography from '@material-ui/core/Typography';
import {
  SPI_PAIR_STATUS,
  TEXT_FORM_CONFIGURATION_AUTO_ADDRESS_VALUE,
  TEXT_FORM_CONFIGURATION_EFTPOS_ADDRESS_VALUE,
  TEXT_FORM_DEFAULT_OPTION,
  TEXT_FORM_DEFAULT_VALUE,
  TEXT_FORM_MODAL_CODE_TILL,
  TEXT_FORM_MODAL_CODE_WESTPAC,
  TEXT_FORM_VALIDATION_EFTPOS_ADDRESS_TEXTFIELD,
  TEXT_FORM_VALIDATION_POS_ID_TEXTFIELD,
  TEXT_FORM_VALIDATION_PROVIDER_TEXTFIELD,
  TEXT_FORM_VALIDATION_SERIAL_NUMBER_TEXTFIELD,
} from '../../../../definitions/constants/commonConfigs';
import { useAppDispatch, useAppSelector } from '../../../../redux/hooks';
import { pairForm, selectPairFormDeviceAddress } from '../../../../redux/reducers/PairFormSlice/PairFormSelectors';
import { updatePairFormParams } from '../../../../redux/reducers/PairFormSlice/pairFormSlice';
import {
  handlePaymentProviderFieldOnChange,
  handlePaymentProviderSelectorOnChange,
  handlePaymentProviderFieldOnBlur,
  handleTestModeCheckboxOnChange,
  handleAddressTypeSelectorOnBlur,
  handleAddressTypeSelectorOnChange,
  handleDeviceAddressFieldOnChange,
  handleDeviceAddressFieldOnBlur,
  handleSerialNumberFieldOnBlur,
  handleSerialNumberFieldOnChange,
  handlePosIdFieldOnBlur,
  handlePosIdFieldOnChange,
} from '../../../../utils/common/pair/pairFormHelpers';
import {
  eftposAddressValidator,
  fieldRequiredValidator,
  serialNumberValidator,
} from '../../../../utils/validators/pairFormValidators';
import CustomTextField from '../../../CustomTextField';
import ErrorInputAdornment from '../../../CustomTextField/ErrorInputAdornment';
import useStyles from '../index.styles';
import { IFormEventCheckbox, IFormEventValue, IPairFormState } from '../interfaces';

export default function PairConfiguration({ terminal }: IPairFormState): React.ReactElement {
  const classes = useStyles();
  const dispatch = useAppDispatch();
  const { acquirerCode, addressType, deviceAddress, posId, serialNumber, testMode } = useAppSelector(pairForm);
  const pairFormDeviceAddress = useAppSelector(selectPairFormDeviceAddress);

  return (
    <>
      <Grid item className={classes.title}>
        <Typography variant="h6" component="h1" className={classes.configurationTitle}>
          Pairing configuration
        </Typography>
      </Grid>
      <Grid container direction="column">
        <Grid container direction="row" className={classes.fieldSpace}>
          <Grid item sm={6} xs={12} className={classes.columnSpace}>
            <FormControl variant="outlined" margin="dense" fullWidth data-test-id="paymentProviderSelector">
              <InputLabel data-test-id="paymentProviderLabel">Payment provider</InputLabel>
              <Select
                className={classes.pairFormSelector}
                data-test-id="paymentProviderSelectorLabel"
                disabled={terminal?.status === SPI_PAIR_STATUS.PairedConnecting}
                label="Payment provider"
                labelId="paymentProviderSelectorLabel"
                onChange={(event: IFormEventValue) =>
                  handlePaymentProviderSelectorOnChange(dispatch, event, fieldRequiredValidator, updatePairFormParams)
                }
                value={acquirerCode.option}
              >
                <MenuItem value={TEXT_FORM_DEFAULT_OPTION}>Payment provider</MenuItem>
                <MenuItem value={TEXT_FORM_MODAL_CODE_TILL}>Till Payments</MenuItem>
                <MenuItem value={TEXT_FORM_MODAL_CODE_WESTPAC}>Westpac</MenuItem>
                <MenuItem value={TEXT_FORM_DEFAULT_VALUE}>Other</MenuItem>
              </Select>
            </FormControl>
          </Grid>
          <Grid item sm={6} xs={12}>
            <CustomTextField
              dataTestId="paymentProviderField"
              disabled={
                acquirerCode.option !== TEXT_FORM_DEFAULT_VALUE || terminal?.status === SPI_PAIR_STATUS.PairedConnecting
              }
              error={!acquirerCode.isValid}
              fullWidth
              helperText={!acquirerCode.isValid ? TEXT_FORM_VALIDATION_PROVIDER_TEXTFIELD : ''}
              InputProps={{
                endAdornment: <ErrorInputAdornment isValid={!acquirerCode.isValid} />,
              }}
              label={TEXT_FORM_DEFAULT_VALUE}
              margin="dense"
              onBlur={(event: IFormEventValue) =>
                handlePaymentProviderFieldOnBlur(dispatch, event, fieldRequiredValidator, updatePairFormParams)
              }
              onChange={(event: IFormEventValue) =>
                handlePaymentProviderFieldOnChange(dispatch, event, updatePairFormParams)
              }
              value={acquirerCode.value}
              variant="outlined"
            />
          </Grid>
        </Grid>
        <Grid container direction="row" className={classes.fieldSpace}>
          <Grid item sm={6} xs={12} className={classes.columnSpace}>
            <FormControl variant="outlined" margin="dense" fullWidth data-test-id="configurationTypeSelector">
              <InputLabel data-test-id="configurationLabel">Configuration option</InputLabel>
              <Select
                className={classes.pairFormSelector}
                data-test-id="configurationSelector"
                disabled={terminal?.status === SPI_PAIR_STATUS.PairedConnecting}
                label="Configuration option"
                labelId="configurationDropdownLabel"
                onBlur={() => handleAddressTypeSelectorOnBlur(dispatch, testMode, updatePairFormParams)}
                onChange={(event: IFormEventValue) =>
                  handleAddressTypeSelectorOnChange(dispatch, event, updatePairFormParams)
                }
                value={addressType}
              >
                <MenuItem value={TEXT_FORM_CONFIGURATION_AUTO_ADDRESS_VALUE}>Auto address</MenuItem>
                <MenuItem value={TEXT_FORM_CONFIGURATION_EFTPOS_ADDRESS_VALUE}>EFTPOS address</MenuItem>
              </Select>
            </FormControl>
          </Grid>
          <Grid item sm={6} xs={12}>
            <CustomTextField
              dataTestId="eftposAddressField"
              disabled={
                addressType === TEXT_FORM_CONFIGURATION_AUTO_ADDRESS_VALUE ||
                terminal?.status === SPI_PAIR_STATUS.PairedConnecting
              }
              error={!deviceAddress.isValid}
              fullWidth
              helperText={!deviceAddress.isValid ? TEXT_FORM_VALIDATION_EFTPOS_ADDRESS_TEXTFIELD : ''}
              InputProps={{
                endAdornment: <ErrorInputAdornment isValid={!deviceAddress.isValid} />,
              }}
              label="EFTPOS address"
              margin="dense"
              onBlur={(event: IFormEventValue) =>
                handleDeviceAddressFieldOnBlur(
                  addressType,
                  dispatch,
                  event,
                  eftposAddressValidator,
                  updatePairFormParams
                )
              }
              onChange={(event: IFormEventValue) =>
                handleDeviceAddressFieldOnChange(dispatch, event, updatePairFormParams)
              }
              value={
                (addressType === TEXT_FORM_CONFIGURATION_AUTO_ADDRESS_VALUE &&
                  terminal?.status === SPI_PAIR_STATUS.PairedConnecting) ||
                terminal?.status === SPI_PAIR_STATUS.PairedConnected
                  ? pairFormDeviceAddress
                  : deviceAddress.value
              }
              variant="outlined"
            />
          </Grid>
        </Grid>
        <Grid item className={classes.fieldSpace}>
          <CustomTextField
            dataTestId="serialNumberField"
            disabled={terminal?.status === SPI_PAIR_STATUS.PairedConnecting}
            error={!serialNumber.isValid}
            fullWidth
            helperText={!serialNumber.isValid ? TEXT_FORM_VALIDATION_SERIAL_NUMBER_TEXTFIELD : ''}
            InputProps={{
              endAdornment: <ErrorInputAdornment isValid={!serialNumber.isValid} />,
            }}
            label="Serial number"
            margin="dense"
            onBlur={(event: IFormEventValue) =>
              handleSerialNumberFieldOnBlur(dispatch, event, serialNumberValidator, updatePairFormParams)
            }
            onChange={(event: IFormEventValue) =>
              handleSerialNumberFieldOnChange(dispatch, event, updatePairFormParams)
            }
            required
            value={serialNumber.value}
            variant="outlined"
          />
        </Grid>
        <Grid item className={classes.fieldSpace}>
          <CustomTextField
            dataTestId="posIdField"
            disabled={terminal?.status === SPI_PAIR_STATUS.PairedConnecting}
            error={!posId.isValid}
            fullWidth
            helperText={!posId.isValid ? TEXT_FORM_VALIDATION_POS_ID_TEXTFIELD : ''}
            InputProps={{
              endAdornment: <ErrorInputAdornment isValid={!posId.isValid} />,
            }}
            label="POS ID"
            margin="dense"
            onBlur={(event: IFormEventValue) =>
              handlePosIdFieldOnBlur(dispatch, event, fieldRequiredValidator, updatePairFormParams)
            }
            onChange={(event: IFormEventValue) => handlePosIdFieldOnChange(dispatch, event, updatePairFormParams)}
            required
            value={posId.value}
            variant="outlined"
          />
        </Grid>
        <Grid item className={classes.fieldSpace}>
          <FormControlLabel
            control={
              <Checkbox
                checked={testMode}
                color="primary"
                disabled={terminal?.status === SPI_PAIR_STATUS.PairedConnecting}
                data-test-id="testModeCheckbox"
                name="testMode"
                onChange={(event: IFormEventCheckbox) =>
                  handleTestModeCheckboxOnChange(dispatch, event, updatePairFormParams)
                }
              />
            }
            label="Test mode"
          />
        </Grid>
      </Grid>
    </>
  );
}
