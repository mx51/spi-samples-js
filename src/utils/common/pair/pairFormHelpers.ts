import { IFormEventCheckbox, IFormEventValue } from '../../../components/PairPage/PairForm/interfaces';
import {
  TEXT_FORM_CONFIGURATION_AUTO_ADDRESS_VALUE,
  TEXT_FORM_CONFIGURATION_EFTPOS_ADDRESS_VALUE,
  TEXT_FORM_DEFAULT_OPTION,
  TEXT_FORM_DEFAULT_VALUE,
} from '../../../definitions/constants/commonConfigs';

import { IUpdatePairFormParams } from '../../../redux/reducers/PairFormSlice/interfaces';
import { ITerminalState } from '../../../redux/reducers/TerminalSlice/interfaces';
import { serialNumberValidatorOnBlur, serialNumberValidatorOnChange } from '../../validators/pairFormValidators';

export function isHttps(): boolean {
  return window.location.protocol === 'https:';
}

export function serialNumberFormatter(currentSerialNumber: string): string {
  let formatSerialNumber = currentSerialNumber.replaceAll('-', '');

  if (formatSerialNumber.length > 3 && formatSerialNumber.length <= 6)
    formatSerialNumber = `${formatSerialNumber.slice(0, 3)}-${formatSerialNumber.slice(3)}`;
  else if (currentSerialNumber.length > 6)
    formatSerialNumber = `${formatSerialNumber.slice(0, 3)}-${formatSerialNumber.slice(
      3,
      6
    )}-${formatSerialNumber.slice(6)}`;

  return formatSerialNumber;
}

export const initialSpiFormData = {
  provider: {
    isValid: true,
    option: TEXT_FORM_DEFAULT_OPTION,
    value: '',
  },
  configuration: {
    isValid: true,
    type: isHttps() ? TEXT_FORM_CONFIGURATION_AUTO_ADDRESS_VALUE : TEXT_FORM_CONFIGURATION_EFTPOS_ADDRESS_VALUE,
    value: '',
  },
  serialNumber: {
    isValid: true,
    value: '',
  },
  posId: {
    isValid: true,
    value: '',
  },
  testMode: !isHttps(),
};

// Payment Provider Selector
export const handlePaymentProviderSelectorOnChange = (
  dispatch: Any,
  event: IFormEventValue,
  fieldRequiredValidator: (value: string) => boolean,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  if (event.target.value !== TEXT_FORM_DEFAULT_OPTION && event.target.value !== TEXT_FORM_DEFAULT_VALUE) {
    dispatch(
      updatePairFormParams({
        key: 'acquirerCode',
        value: {
          value: event.target.value as string,
          option: event.target.value as string,
          isValid: fieldRequiredValidator(event.target.value as string),
        },
      })
    );
  } else {
    dispatch(
      updatePairFormParams({
        key: 'acquirerCode',
        value: {
          value: '',
          option: event.target.value as string,
          isValid: false,
        },
      })
    );
  }
};

// Payment Provider Field
export const handlePaymentProviderFieldOnChange = (
  dispatch: Any,
  event: IFormEventValue,
  paymentProviderValidator: (value: string) => boolean,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'acquirerCode',
      value: {
        value: event.target.value as string,
        option: TEXT_FORM_DEFAULT_VALUE,
        isValid: paymentProviderValidator(event.target.value as string),
      },
    })
  );
};

export const handlePaymentProviderFieldOnBlur = (
  dispatch: Any,
  event: IFormEventValue,
  paymentProviderValidator: (value: string) => boolean,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'acquirerCode',
      value: {
        value: event.target.value as string,
        option: TEXT_FORM_DEFAULT_VALUE,
        isValid: paymentProviderValidator(event.target.value as string),
      },
    })
  );
};

// Address Type Selector
export const handleAddressTypeSelectorOnChange = (
  dispatch: Any,
  event: IFormEventValue,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'addressType',
      value: event.target.value as string,
    })
  );

  dispatch(
    updatePairFormParams({
      key: 'deviceAddress',
      value: {
        value: '',
        isValid: true,
      },
    })
  ); // reset the device address field value when address type change detected
};

export const handleAddressTypeSelectorOnBlur = (
  dispatch: Any,
  checkedValue: boolean,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'testMode',
      value: !checkedValue,
    })
  );
};

// Device Address Field
export const handleDeviceAddressFieldOnChange = (
  dispatch: Any,
  event: IFormEventValue,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'deviceAddress',
      value: {
        value: event.target.value as string,
        isValid: true,
      },
    })
  );
};

export const handleDeviceAddressFieldOnBlur = (
  addressType: string,
  dispatch: Any,
  event: IFormEventValue,
  eftposAddressValidator: (type: string, value: string) => boolean,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'deviceAddress',
      value: {
        value: event.target.value as string,
        isValid: eftposAddressValidator(addressType, event.target.value as string),
      },
    })
  );
};

// Serial Number Field
export const handleSerialNumberFieldOnChange = (
  dispatch: Any,
  event: IFormEventValue,
  serialNumberValidator: (value: string) => string,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  const currentSerialNumber = (event.target.value as unknown as string).slice(0, 11);

  dispatch(
    updatePairFormParams({
      key: 'serialNumber',
      value: {
        value: serialNumberFormatter(currentSerialNumber),
        isValid: serialNumberValidatorOnChange(event.target.value as string) === '',
      },
    })
  );
};

export const handleSerialNumberFieldOnBlur = (
  dispatch: Any,
  event: IFormEventValue,
  serialNumberValidator: (value: string) => string,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'serialNumber',
      value: {
        value: event.target.value as string,
        isValid: serialNumberValidatorOnBlur(event.target.value as string) === '',
      },
    })
  );
};

// POS ID Field
export const handlePosIdFieldOnChange = (
  dispatch: Any,
  event: IFormEventValue,
  fieldRequiredValidator: (value: string, terminals: ITerminalState) => string,
  terminals: ITerminalState,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'posId',
      value: {
        value: event.target.value as string,
        isValid: fieldRequiredValidator(event.target.value as string, terminals) === '',
      },
    })
  );
};

export const handlePosIdFieldOnBlur = (
  dispatch: Any,
  event: IFormEventValue,
  fieldRequiredValidator: (value: string, terminals: ITerminalState) => string,
  terminals: ITerminalState,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'posId',
      value: {
        value: event.target.value as string,
        isValid: fieldRequiredValidator(event.target.value as string, terminals) === '',
      },
    })
  );
};

// Test Mode Checkbox
export const handleTestModeCheckboxOnChange = (
  dispatch: Any,
  event: IFormEventCheckbox,
  updatePairFormParams: IUpdatePairFormParams
): void => {
  dispatch(
    updatePairFormParams({
      key: 'testMode',
      value: event.target.checked,
    })
  );
};
