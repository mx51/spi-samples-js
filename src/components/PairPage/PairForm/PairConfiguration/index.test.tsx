import React from 'react';
import { cleanup } from '@testing-library/react';
import PairConfiguration from '.';
import { initialSpiFormData } from '../../../../utils/common/pair/pairFormHelpers';
import mockWithRedux, { defaultMockTerminals, mockTerminalInstanceId } from '../../../../utils/tests/common';

describe('Test <PairConfiguration />', () => {
  afterEach(cleanup);

  test('should match PairConfiguration snapshot test', () => {
    // Arrange
    const setSpi = jest.fn();
    const terminal = defaultMockTerminals[mockTerminalInstanceId];
    const container = mockWithRedux(<PairConfiguration setSpi={setSpi} spi={initialSpiFormData} terminal={terminal} />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
