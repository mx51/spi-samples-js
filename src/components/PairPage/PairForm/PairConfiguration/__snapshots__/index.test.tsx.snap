// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PairConfiguration /> should match PairConfiguration snapshot test 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-title-3 MuiGrid-item"
  >
    <h1
      class="MuiTypography-root makeStyles-configurationTitle-4 MuiTypography-h6"
    >
      Pairing configuration
    </h1>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root makeStyles-fieldSpace-5 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-columnSpace-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth"
          data-test-id="paymentProviderSelector"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            data-test-id="paymentProviderLabel"
          >
            Payment provider
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-pairFormSelector-11 MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            data-test-id="paymentProviderSelectorLabel"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="paymentProviderSelectorLabel"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              role="button"
              tabindex="0"
            >
              Payment provider
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="Payment provider"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
              >
                <span>
                  Payment provider
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-20 MuiFormControl-marginDense MuiFormControl-fullWidth"
          data-test-id="paymentProviderField"
          theme="[object Object]"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled"
            data-shrink="false"
          >
            (Other) Please specify
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              disabled=""
              type="text"
              value=""
            />
            
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18"
              >
                <span>
                  (Other) Please specify
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-fieldSpace-5 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-columnSpace-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth"
          data-test-id="configurationTypeSelector"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            data-test-id="configurationLabel"
          >
            Configuration option
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root makeStyles-pairFormSelector-11 MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            data-test-id="configurationSelector"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="configurationDropdownLabel"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              role="button"
              tabindex="0"
            >
              EFTPOS address
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="eftpos"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
              >
                <span>
                  Configuration option
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-20 MuiFormControl-marginDense MuiFormControl-fullWidth"
          data-test-id="eftposAddressField"
          theme="[object Object]"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            EFTPOS address
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              type="text"
              value=""
            />
            
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18"
              >
                <span>
                  EFTPOS address
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-fieldSpace-5 MuiGrid-item"
    >
      <div
        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-20 MuiFormControl-marginDense MuiFormControl-fullWidth"
        data-test-id="serialNumberField"
        theme="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          Serial number
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
            â€‰
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            required=""
            type="text"
            value=""
          />
          
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-18"
            >
              <span>
                Serial number
                Â *
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-fieldSpace-5 MuiGrid-item"
    >
      <div
        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-20 MuiFormControl-marginDense MuiFormControl-fullWidth"
        data-test-id="posIdField"
        theme="[object Object]"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
        >
          POS ID
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
            â€‰
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            maxlength="16"
            required=""
            type="text"
            value=""
          />
          
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-18"
            >
              <span>
                POS ID
                Â *
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-fieldSpace-5 MuiGrid-item"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-22 Mui-checked MuiIconButton-colorPrimary"
          data-test-id="testModeCheckbox"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-24"
              data-indeterminate="false"
              name="testMode"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Test mode
        </span>
      </label>
    </div>
  </div>
</div>
`;
